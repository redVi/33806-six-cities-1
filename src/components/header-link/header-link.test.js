import React from 'react';
import {MemoryRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import HeaderLink from '@/components/header-link/header-link.jsx';

describe(`HeaderLink`, () => {
  it(`renders correctly when user isn't logged in`, () => {
    const tree = renderer.create(
        <Router>
          <HeaderLink link="/login" text="Sign In" />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly when user is logged in`, () => {
    const tree = renderer.create(
        <Router>
          <HeaderLink
            link="/favorites"
            text="jack-sparrow@pirates.com"
            linkClass="user-name user__name"
            bg={{backgroundImage: `url("https://pirates.com/uploads/avatar.jpg")`}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
