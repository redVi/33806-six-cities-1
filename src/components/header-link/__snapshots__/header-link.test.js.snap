// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLink renders correctly when user is logged in 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/favorites"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
    style={
      Object {
        "backgroundImage": "url(\\"https://pirates.com/uploads/avatar.jpg\\")",
      }
    }
  />
  <span
    className="header__user-name user__name"
  >
    jack-sparrow@pirates.com
  </span>
</a>
`;

exports[`HeaderLink renders correctly when user isn't logged in 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="/login"
  onClick={[Function]}
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />
  <span
    className="header__login"
  >
    Sign In
  </span>
</a>
`;
